File: mcdowell/js/ch18-02.js
function shuffleDeck(inputs) {
    for(var i = 0; i < inputs.length; i ++) {
        var k = Math.floor(Math.random() * (i + 1));
        var tmp = inputs[k];
        inputs[k] = inputs[i];
        inputs[i] = tmp;
    }

    return inputs;
}

for (var i = 0; i < 100; i ++) {
    console.log(shuffleDeck([1,2,3,4,5]));
}

File: mcdowell/python/ch18-02.py
import random

def shuffleDeck(inputs):

    for i in range(len(inputs)):
        k = random.randint(0, i)
        # print k
        tmp = inputs[k]
        inputs[k] = inputs[i]
        inputs[i] = tmp

for i in range(100):
    a = [1,2,3,4,5,6,7]
    shuffleDeck(a)
    print a